### PR TITLE
Feature/default co2 rf method

### LINKED ISSUE #43: Default CO2 RF method
"Etminan 2016" method as default method to calculate CO2 RF unless specified otherwise by the user.

### PR DESCRIPTION
**Pull request by stefan-voelk**
## Before you get started
- [x] 🙋‍♀️ Create an issue to discuss what you are going to do!

---

## Description
- Closes #43 

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. 

## Note
- This feature goes beyond defining a default CO2 RF method: Any other settings can be included in DEFAULT_CONFIG within module read_config. These default settings will be added to config if not specified by user in config file. Additionally, a logging message is output to inform user which default settings are used.
- Added new dependency, python package "deepmerge" for merging nested dictionaries, e.g. config and added_dict.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change

## How has this been tested?
Please describe the software tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Update of tests inTestCheckConfig
- [x] Add TestGetKeysValues
- [x] Add TestAddDefaultConfig

**Test configuration**:
* Operating system: local machine: Windows 11 Enterprise (64-Bit)
* Other relevant information:

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any changed dependencies have been added or removed correctly

### CODE REVIEW

File: tests/read_config_test.py [hunk 1]
```diff
@@ -37,18 +38,47 @@ def test_invalid(self):
...
+@pytest.fixture(name="setup_arguments", scope="class")
+def fixture_setup_arguments():
+    """Setup arguments for check_config
+
+    Returns:
+        dict, dict: Configuration template and default config
+    """
+    config_template = {
+        "species": {"inv": Iterable, "out": Iterable},
+        "inventories": {"dir": str, "files": Iterable},
```
Comment (liammegill) [inline tests/read_config_test.py:50]: Here we also need to add `inventories.rel_to_base`

File: openairclim/read_config.py [hunk 2]
```diff
@@ -8,12 +8,14 @@
...
+from deepmerge import Merger
 import numpy as np
 
 # CONSTANTS
-# List types of all mandatory config parameters
-CONFIG_TYPES = {
+# Template of config dictionary with types of MANDATORY input settings
+CONFIG_TEMPLATE = {
     "species": {"inv": Iterable, "out": Iterable},
     "inventories": {"dir": str, "files": Iterable},
```
Comment (liammegill) [inline openairclim/read_config.py:20]: `inventories.rel_to_base` must also be defined (bool).

**Review by liammegill — COMMENTED** [review]
Looks good! Just a few minor comments. Whilst doing this review with a clean installation of OpenAirClim, I noticed that a number of the tests did not pass, which I now found out to be because they require `create_test_files.py` to have been run. I opened issue #50 to address this.

File: tests/read_config_test.py (hunk 1 above)
Comment (stefan-voelk) [reply tests/read_config_test.py:50]: added

File: example/example.toml
[the remark this answers is not in the document]
Comment (stefan-voelk) [reply example/example.toml:1]: comment added

**Review by liammegill — APPROVED** [review]
Looks good to me!

**Pull request merged**